symfit/core/fit.py

```diff
@@ -1753,33 +1749,53 @@ def eval_components(self, *args, **kwargs):
         Dfun = lambda ys, t, *a: [[c(t, *(list(ys) + list(a))) for c in row] for row in self._njacobian]
 
         initial_dependent = [self.initial[var] for var in self.dependent_vars]
-        initial_independent = self.initial[self.independent_vars[0]] # Assuming there's only one
+        t_initial = self.initial[self.independent_vars[0]] # Assuming there's only one
 
         # Check if the time-like data includes the initial value, because integration should start there.
-        # For fitting to make sence, it should probably not be in there though. Needs mathematical backing.
         try:
             t_like[0]
         except (TypeError, IndexError): # Python scalar gives TypeError, numpy scalars IndexError
             t_like = np.array([t_like]) # Allow evaluation at one point.
 
-        if t_like[0] == initial_independent:
-            start = 0
-            warnings.warn("The initial point should probably not be included with your data points as this point will always be fitted perfectly.")
-        elif t_like[0] < initial_independent:
-            raise ModelError('ODEModel\'s can not be evaluated for values smaller than the initial value')
+        # The strategy is to split the time axis in a part above and below the
+        # initial value, and to integrate those seperately. At the end we rejoin them.
+        # np.flip is needed because odeint wants the first point to be t_initial
+        if t_initial in t_like:
+            t_bigger = t_like[t_like >= t_initial]
+            t_smaller = np.flip(t_like[t_like <= t_initial], axis=-1)
         else:
-            assert len(t_like.shape) == 1
-            t_like = np.hstack((np.array([initial_independent]), t_like))
-            start = 1
-        ans = odeint(
+            t_bigger = np.concatenate(
+                (np.array([t_initial]), t_like[t_like > t_initial])
+            )
+            t_smaller = np.concatenate(
+                (np.array([t_initial]), np.flip(t_like[t_like < t_initial], axis=-1))
+            )
+            t_total = np.concatenate((t_like[t_like < t_initial] ,t_bigger))
```

Comment:
Why not `t_total = np.concatenate((t_smaller[:-1], t_bigger))`?
And this line should probably be dedented one tick.

Reply:
changed it to `t_total = np.concatenate((t_smaller[::-1][:-1], t_bigger))`